core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,6 +107,19 @@ public static Snapshot oldestAncestor(Table table) {
     return lastSnapshot;
   }
 
+  /**
+   * Traverses the history and finds the oldest ancestor of snapshotId.
```

Comment:
Note that this may return `snapshotId` if that snapshot is the oldest known snapshot (its parent is null or has been expired).